modules/gateway/nodes.go

```diff
@@ -93,13 +93,15 @@ func (g *Gateway) relayNode(conn modules.PeerConn) error {
 	}
 	// add node
 	id := g.mu.Lock()
-	defer g.mu.Unlock(id)
 	if err := g.addNode(addr); err != nil {
+		g.mu.Unlock(id)
 		return err
 	}
 	g.save()
+	g.mu.Unlock(id)
```

Comment:
I don't see the harm, other than performance. Luke and I were discussing reducing the total save volume by as much as possible to help with performance, Sia currently goes to disk all the time.